rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -1203,25 +1319,36 @@ def _prepare_sparse_dense_layers(
                 )
 
     def _prepare_input_layers(self, name: Text) -> None:
-        self._tf_layers[f"sparse_input_dropout.{name}"] = layers.SparseDropout(
-            rate=self.config[DROP_RATE]
-        )
-        self._tf_layers[f"dense_input_dropout.{name}"] = tf.keras.layers.Dropout(
-            rate=self.config[DROP_RATE]
-        )
-        self._prepare_sparse_dense_layers(
-            self.data_signature[f"{name}_features"],
-            name,
-            self.config[REGULARIZATION_CONSTANT],
-            self.config[DENSE_DIMENSION][name],
-        )
         self._tf_layers[f"ffnn.{name}"] = layers.Ffnn(
             self.config[HIDDEN_LAYERS_SIZES][name],
             self.config[DROP_RATE],
             self.config[REGULARIZATION_CONSTANT],
             self.config[WEIGHT_SPARSITY],
             name,
         )
+        for feature_type in [SENTENCE, SEQUENCE]:
+            if f"{name}_{feature_type}_features" not in self.data_signature:
+                continue
+
+            self._tf_layers[
+                f"sparse_input_dropout.{name}_{feature_type}"
+            ] = layers.SparseDropout(rate=self.config[DROP_RATE])
+            self._tf_layers[
+                f"dense_input_dropout.{name}_{feature_type}"
+            ] = tf.keras.layers.Dropout(rate=self.config[DROP_RATE])
+            self._prepare_sparse_dense_layers(
+                self.data_signature[f"{name}_{feature_type}_features"],
+                f"{name}_{feature_type}",
+                self.config[REGULARIZATION_CONSTANT],
+                self.config[DENSE_DIMENSION][name],
+            )
+            self._tf_layers[f"ffnn.{name}_{feature_type}"] = layers.Ffnn(
```

Comment:
I think the name of the layer `ffnn` is confusing here, because it clashes with ffnn before transformer

Reply:
Should not clash: We first have `ffnn.text_sequence` and `ffnn.text_sentence` and then right before the transformer we have `ffnn.text`. Do you think we should add something to the name?

Reply:
I meant confusing from a reader of the code point of view, I would call it `concat_layer` and use a `Dense` layer instead of `Ffnn` one, so that the name explicitly state its purpose